bbb/services.py

```diff
@@ -285,6 +285,16 @@ def reflectTasks(self):
             # If takenUntil isn't set, this task has either never been claimed
             # or got cancelled.
             if not t.takenUntil:
```

Comment:
I wonder if we should do, `if not t.takenUntil and not arrow.now() < arrow.get(t.processedDate).replace(minutes=5):`

so then we end up here: https://github.com/mozilla/buildbot-bridge/blob/ed43e28e0e6041691c7bcc4866cd0c788a24df1c/bbb/services.py#L312

 which gives us the same logic flow as what you are doing in this patch without the added complexity.